samples/constitutions/default/actions.js

```diff
@@ -762,6 +762,25 @@ const actions = new Map([
       }
     ),
   ],
+  [
+    "set_js_engine_options",
+    new Action(
+      function (args) {
+        checkType(args.max_heap_bytes, "integer", "max_heap_bytes");
+        checkType(args.max_stack_bytes, "integer", "max_stack_bytes");
+      },
+      function (args) {
+        const js_engine_map = ccf.kv["public:ccf.gov.jsengine"];
+        const heap_bytes = args.max_heap_bytes;
+        const stack_bytes = args.max_stack_bytes;
+        const heap_name = ccf.strToBuf("max_heap_bytes");
+        const stack_name = ccf.strToBuf("max_stack_bytes");
+        js_engine_map.set(heap_name, ccf.jsonCompatibleToBuf(heap_bytes));
+        js_engine_map.set(stack_name, ccf.jsonCompatibleToBuf(stack_bytes));
```

Comment:
We write these values to the new `jsengine` KV map but this isn't actually enforced by the fact that these are written to the map but via a different call to `updateJSRuntimeMemory` below, which is a little odd. 

Usually, values written to the KV that have side effects on the node are picked up via a KV hook (in this case, global). Otherwise, this means that this value is only picked up by the node executing the proposal (here, the primary node).

Reply:
The simplest thing to do would be to read them from the KV at every Runtime instantiation. We already do that for the code/bytecode, we could do that cheaply for those values too.

Hooks-based things are hard to make correct and transactional, and I don't think we especially want to wait for a global hook on this.

Reply:
I just noticed this. (I should have better GitHub notifications?!)

You're right, this change indeed effects only the current executing node. I also need to update the js_generic code as well.

Reply:
Done